GearException.cpp

```diff
@@ -0,0 +1,5 @@
+#include "GearException.hpp"
+
+const char* GearException::what() const noexcept {
+    return msg_.c_str();
+}
```

Comment:
There is no need to override this function. It's exactly the same behavior as in std::exception.